spec/controllers/my/all_controller_spec.rb

```diff
@@ -2,80 +2,59 @@
 
 describe My::AllController, :type => :controller do
 
-#  routes { Dashboard::My::Engine.routes }
+  describe "User" do
+    before(:all) do
+      cleanup_jetty # TODO: Weird dirtying of database from other specs causing the shared_file spec to fail?
+    end
 
-  let(:user) { FactoryGirl.find_or_create(:archivist) }
-  let(:admin) { FactoryGirl.find_or_create(:admin) }
+    let(:user) { FactoryGirl.create(:jill) }
+    let(:other_user) { FactoryGirl.create(:alice) }
 
-  let(:my_file) do
-    GenericFile.new.tap do |g|
-      g.apply_depositor_metadata(admin.user_key)
-      g.save!
+    let!(:my_file) do
+      FactoryGirl.create(:generic_file, depositor: user)
     end
-  end
 
-  let(:my_collection) do
-    Collection.new(title: 'test collection').tap do |c|
-      c.apply_depositor_metadata(user.user_key)
-      c.save!
+    let!(:my_collection) do
+      Collection.new(title: 'test collection') do |c|
+        c.apply_depositor_metadata(user.user_key)
+        c.save!
+      end
     end
-  end
 
-  let(:shared_file) do
-    FactoryGirl.build(:generic_file).tap do |r|
-      r.apply_depositor_metadata FactoryGirl.find_or_create(:user)
-      r.edit_users += [user.user_key]
-      r.save!
+    let!(:shared_file) do
+      FactoryGirl.create(:generic_file, depositor: other_user) do |gf|
+        gf.edit_users += [user.user_key]
+        gf.save!
+      end
     end
-  end
-  
-  before do
-    sign_in user
-    @my_file = my_file
-    @my_collection = my_collection
-    @shared_file = shared_file
-    @unrelated_file = FactoryGirl.create(:generic_file, depositor: FactoryGirl.find_or_create(:user))
-    @wrong_type = Batch.create
-  end
-
-  after do
-    GenericFile.destroy_all
-    Collection.destroy_all
-  end
 
-
-  describe "User" do
-    before do
-      sign_in user
+    let!(:unrelated_file) do
+      FactoryGirl.create(:generic_file, depositor: other_user)
     end
 
-    it 'should respond with error' do
-      expect{ get :index, user_route: dashboard }.to raise_error
```

Comment:
This test isn't testing the right thing. A basic user can visit their /my/all_controller index, you do not need to be an admin to do this.

But most importantly this shows an example of a false positive from using raise_error. This is only passing because we miss spelled `use_route` as `user_route` which isn't valid, and we are capturing this error instead!